packages/framework/src/DependencyInjection/Compiler/RegisterProjectFrameworkClassExtensionsCompilerPass.php

```diff
@@ -44,4 +63,29 @@ private function isProjectClass(string $serviceId): bool
     {
         return strpos($serviceId, 'App') !== false;
     }
+
+    /**
+     * @param \Roave\BetterReflection\Reflection\ReflectionClass $frameworkClassBetterReflection
+     * @param \Symfony\Component\DependencyInjection\Definition $classExtensionRegistryDefinition
+     * @param string $aliasId
+     */
+    private function addAllVariantsOfServiceToClassExtension(
+        ReflectionClass $frameworkClassBetterReflection,
+        Definition $classExtensionRegistryDefinition,
+        string $aliasId
+    ): void {
+        if ($frameworkClassBetterReflection->isInterface()) {
+            $className = rtrim($frameworkClassBetterReflection->getName(), 'Interface');
```

Comment:
rtrim does not work as presumed here. 

for example  
`rtrim('Myface', 'Interface')` results in `My` 
`rtrim('MyFacadeInterface', 'Interface')` result in `MyFacad`

Maybe `preg_replace('/Interface$/', '', $string);` would be a better choice

Reply:
Changed